ztp/gitops-subscriptions/argocd/PowerConfiguration.md

```diff
@@ -0,0 +1,224 @@
+          - "rcupdate.rcu_normal_after_boot=0"
+          - "efi=runtime"
+        ...
+        workloadHints:
+          highPowerConsumption: false
+          perPodPowerManagement: true
+        ...
+    - fileName: TunedPerformancePatch.yaml
+      policyName: "config-policy"
+      spec:
+        profile:
+          - name: performance-patch
+            data: |
+              ...
+              [sysfs]
+              /sys/devices/system/cpu/intel_pstate/max_perf_pct=72
+```
+
+### 4) [Mix of Power saving and Performance mode](#mix-power-performance-mode)
+
+The SNO can be deployed to a default Power saving mode as shown above. For critical workloads that require the highest performance and lowest latency, each such pod should be annotated to disable c-states and set the cpufreq governor to `performance`.
+
+The pod spec should be configured as follows:
+
+```yaml
+metadata:
+  annotations:
+    cpu-c-states.crio.io: "disable"
+    cpu-freq-governor.crio.io: "performance"
+```
+
+It is worthwhile to note the following:
+
+- These annotations are likely to be used with the other high performance cri-o annotation described [here](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/scalability_and_performance/cnf-low-latency-tuning#doc-wrapper).
+- These annotations have the same restrictions as the other high performance cri-o annotations:
+    - The pod must use the performance-&lt;profile-name&gt; runtime class.
+    - The pod must have a [QoS Class of guaranteed](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed) (i.e. uses whole CPUs).
+
+## Migrating from one mode to another
+
+This section outlines the steps and configurations for migrating from one mode to another mode.
+
+### 1) Performance to Power saving
+
+Migrating from a Performance configuration to a Power saving configuration requires setting the workloadHints parameters as follows:
+
+```yaml
+workloadHints:
+  highPowerConsumption: false
+  perPodPowerManagement: true
+```
+
+It is recommended to set a cpu governor in the additional kernel arguments specification as well as capping the max cpu frequency as explained in the [power saving mode](#power-saving-mode).
+
+After the above policy changes have been made, it should be pushed to the Git repository and synced through argocd policy application. If the policy has a `remediationAction` set to `enforce`, then TALM (topology aware lifecycle manager) should do the needful of applying the policy to the effected SNOs. If however the `remediationAction` is configured for `inform`, then a CGU (cluster group upgrade) CR must be created and applied to the hub cluster.
+
+After the policy has been applied to the effected SNOs, the change can be verified by viewing the PerformanceProfile on the SNOs:
+
+```sh
+oc get PerformanceProfile -o yaml --kubeconfig sno_kubeconfig.yaml
```

Comment:
I would leave off the --kubeconfig parameter to avoid confusion (also in the example in the next section).

Reply:
I had added it to emphasize to the readers that the kubeconfig of the SNO in question should be used (and not say the hub kubeconfig).

Reply:
I'd prefer to leave it off but I'm OK either way.

Reply:
I'll remove it :)